graylog2-server/src/main/java/org/graylog2/dashboards/widgets/WidgetPosition.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog2.dashboards.widgets;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class WidgetPosition {
+
+    public abstract String id();
+    public abstract Integer width();
+    public abstract Integer height();
+    public abstract Integer col();
+    public abstract Integer row();
+
+    public static WidgetPosition create(String id, Integer width, Integer height, Integer col, Integer row) {
```

Comment:
Instead of using a static factory method, I would suggest using a builder to create `WidgetPosition` instances. We have four attributes of the same type here, making it prone for mixups. E.g.:

While reading the following usage of the factory method:

`WidgetPosition.create(id, 3, 2, 1, 4)`

Can you tell me which numeric parameter is which, from the top of your head? Or:

`WidgetPosition.create(id, height, width, row, col)`

Can you tell if the parameters are in the right order without looking up the definition?

By using a builder pattern, it would like this:

```
WidgetPosition.builder()
    .height(height)
    .width(width)
    .row(row)
    .column(col)
    .build()
```

Making it possible to understand it right on the spot.